Setup tests environment (#64)
stayawayinesctec/stayaway-app: README.md

```sh
$ yarn
```

- If you're using **android**, you'll also need to manually download the binary distribution of [`play-services-nearby`](https://github.com/google/exposure-notifications-android/raw/7dc2979051703443895123f83c370f64a81baa38/app/libs/play-services-nearby-18.0.3-eap.aar) and attach it to your project on `android/app/libs`. You can do so by running on your project's root:
```sh
$ wget https://github.com/google/exposure-notifications-android/raw/7dc2979051703443895123f83c370f64a81baa38/app/libs/play-services-nearby-18.0.3-eap.aar -P android/app/libs
```

## Building
- And finally, build the project:
```sh
$ yarn android # or yarn android:ui
$ yarn ios # or yarn ios:ui
```

NOTE: Running `android:ui` or `ios:ui` will allow you to run and test the applications' user interface without requiring to activate the Exposure Notification API.

## Tests
Once you're ready, you can run the app tests. The tests structure and adopted technologies were inspired by this React Native website's [post](https://reactnative.dev/docs/testing-overview).

### Unit Tests
For unit testing we consider all tests made to one or more individual modules to see if they work as expected.

> The choosed technology was [JEST](https://github.com/facebook/jest).

Reviewer: ```suggestion
For this we use [JEST](https://github.com/facebook/jest).
```

To run unit tests use the following command:
```sh
$ yarn test:unit
```

### Integration Tests
For integration testing we consider all tests made to individual modules combined to see if their cooperation works as expected.

> The choosed technology was [JEST](https://github.com/facebook/jest).

Reviewer: ```suggestion
For this we use [JEST](https://github.com/facebook/jest).
```

To run unit tests use the following command:
```sh
$ yarn test:integration
```

### Component Tests

> For components testing we consider all tests made to the components render to check if they render as expected and respond to every user's interaction.

Reviewer: ```suggestion
For component testing, test will pass if screens render as expected and respond to user interaction.
```

> The choosed technology was [React Native Testing Library](https://github.com/callstack/react-native-testing-library).

Reviewer: ```suggestion
To build these tests we rely on [React Native Testing Library](https://github.com/callstack/react-native-testing-library).
```

To run unit tests use the following command:
```sh
$ yarn test:components
```

### End-to-End

> For e2e testing we consider tests made to the app as an all. In these tests we usually tests all important app flows, as onboarding, submiting a diagnosis, changing the infection status, etc. These tests are running on an emulator/simulator which do not have access to exposure notifications api, therefore we use the UI app's version.

Reviewer: ```suggestion
The e2e test aim at testing the critical user path within the app. Critical app flows such as onboarding, users marking themselves as infected, and risky contact notification are the main focus here. These tests need to run on an emulator/simulator which does not have access to exposure notifications APIs thus requiring the use of the UI-only app version.
```

> The choosed technology was [Detox](https://github.com/wix/Detox).

Reviewer: ```suggestion
The framework chosen to support these tests was [Detox](https://github.com/wix/Detox).
```

> To run unit tests, you need to step your tests environmnet, and you do so by carefully go through every step on the setup tests environment guide ([iOS](https://github.com/wix/Detox/blob/master/docs/Introduction.IosDevEnv.md)/[Android](https://github.com/wix/Detox/blob/master/docs/Introduction.AndroidDevEnv.md)).

Reviewer: ```suggestion
To run tests, you need to step your tests environmnet, and you do so by carefully go through every step on the setup tests environment guide ([iOS](https://github.com/wix/Detox/blob/master/docs/Introduction.IosDevEnv.md)/[Android](https://github.com/wix/Detox/blob/master/docs/Introduction.AndroidDevEnv.md)).
```
